File: Frontend/models/getter/group/GroupItemViewModel.ts
import { ItemStatus } from '../../enums/ItemStatus';
import GroupViewModel from './GroupViewModel';

export default class GroupItemViewModel {
  public Id: number;
  public Name: string;
  public ItemStatusId: ItemStatus;
  public LastUpdate: string;
  public Notes: string = '';
  public DoNotBuy: boolean;
  public OneTimePurchase: boolean;
  public DeletionLoading: boolean = false;
  public Group: GroupViewModel;

  constructor(data?: any, group?: GroupViewModel) {
    if (!data) return;
    this.Id = data.id || data.itemId;
    this.Name = data.name;
    this.ItemStatusId = data.itemStatusId;
    this.LastUpdate = data.lastUpdate;
    this.Notes = data.notes;
    this.DoNotBuy = data.doNotBuy;
    this.OneTimePurchase = data.oneTimePurchase;
    if (group)
      this.Group = group;
  }
}

File: Backend/Application/Groups/Queries/GroupDetail/GroupItemViewModel.cs
using System;
using Domain.Enums;
using Domain.Models;

namespace Application.Groups.Queries.GroupDetail
{
	public class GroupItemViewModel
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public ItemStatusEnum ItemStatusId { get; set; }
		public DateTime LastUpdate { get; set; }
		public string Notes { get; set; }
		public bool DoNotBuy { get; set; }
		public bool OneTimePurchase { get; set; }

		public GroupItemViewModel(GroupItem groupItem)
		{
			Id = groupItem.ItemId;
			Name = groupItem.Item.Name;
			ItemStatusId = groupItem.ItemStatusId;
			LastUpdate = groupItem.LastUpdate;
			Notes = groupItem.Notes;
			DoNotBuy = groupItem.DoNotBuy == true;
			OneTimePurchase = groupItem.OneTimePurchase == true;
		}
	}
}